### ISSUE TITLE
Multiple `World` fails

### ISSUE THREAD

**Issue by evergreen-trading-systems** [cucumber-rs/cucumber#148]
Please add some gotchas or common pitfalls to docs.

FAILS `conflicting implementation for CucumberThenRegexAsync`
```rust
#[derive(Debug, WorldInit)]
pub struct AnimalWorld1 {
  
}

#[async_trait(?Send)]
impl World for AnimalWorld1 {
    type Error = Infallible;
    async fn new() -> Result<Self, Infallible> {
        Ok(Self {
       
        })
    }
}
#[derive(Debug, WorldInit)]
pub struct AnimalWorld2 {
  
}

#[async_trait(?Send)]
impl World for AnimalWorld2 {
    type Error = Infallible;
    async fn new() -> Result<Self, Infallible> {
        Ok(Self {
       
        })
    }
}
```

**Comment by ilslv** [comment]
@evergreen-trading-systems actually proc-macros where designed with this feature in mind. But `WorldInit` produces structs with the same names. Fixed and covered with tests to avoid regression in #150

For now you can place your `World` declarations in separate modules and re-export them
```rust
pub use self::{first::AnimalWorld1, second::AnimalWorld2};

pub mod first {
    #[derive(Debug, WorldInit)]
    pub struct AnimalWorld1;

    #[async_trait(?Send)]
    impl World for AnimalWorld1 {
        type Error = Infallible;
        async fn new() -> Result<Self, Infallible> {
            Ok(Self {
        
            })
        }
    }
}

pub mod second {
    #[derive(Debug, WorldInit)]
    pub struct AnimalWorld2;

    #[async_trait(?Send)]
    impl World for AnimalWorld2 {
        type Error = Infallible;
        async fn new() -> Result<Self, Infallible> {
            Ok(Self {
        
            })
        }
    }
}
```

**Comment by tyranron** [comment]
@evergreen-trading-systems fixed in #150 and released in `0.10.2`.
